templates/main.bicep

```diff
@@ -0,0 +1,358 @@
+param odhsiWebApiName string = 'ohdsi-webapi'
+param dockerRegistryServer string = 'https://index.docker.io/v1'
+param dockerImageName string = 'ohdsi/webapi'
+param dockerImageTag string = '2.12.1'
+param keyvaultName string ='keyvault-${uniqueString(resourceGroup().id)}'
+
+// Postgres parameters
+param serverEdition string = 'GeneralPurpose'
+param skuSizeGB int = 32
+param dbInstanceType string = 'Standard_D2s_v3'
+param availabilityZone string = '3'
+param version string = '14'
+param postgresDatabaseName string = 'atlas_db'
+param postgresSchemaName string = 'webapi'
+param flywayBaselineVersion string = '2.2.5.20180212152023'
+// Users and passwords
+param postgresAdminUsername string = 'postgres_admin'
+param postgresWebapiAdminUsername string = 'ohdsi_admin_user'
+param postgresWebapiAppUsername string = 'ohdsi_app_user'
+@secure()
+param postgresAdminPassword string
+@secure()
+param postgresWebapiAdminPassword string
+@secure()
+param postgresWebapiAppPassword string
+// Connection strings
+var jdbcConnectionStringWebapiAdmin = 'jdbc:postgresql://${postgresServer.properties.fullyQualifiedDomainName}:5432/${postgresDatabaseName}?user=${postgresWebapiAdminUsername}&password=${postgresWebapiAdminPassword}&sslmode=require'
+
+// User Assigned Identity
+resource identity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  name: 'identity-${odhsiWebApiName}'
+  location: location
+}
+
+// KeyVault and storing PostgreSql secrets
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
+  name: keyvaultName
+  location: location
+  properties: {
+    accessPolicies:[
+      {
+        objectId: identity.properties.principalId
+        permissions: {
+          secrets: [
+            'Get'
+            'List'
+          ]
+        }
+        tenantId: tenantId
+      }
+    ]
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: tenantId
+    networkAcls: {
+      defaultAction: 'Allow'
+      bypass: 'AzureServices'
+    }
+  }
+}
+
+resource postgresAdminSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: 'postgres-admin-password'
+  parent: keyVault
+  properties: {
+    value: postgresAdminPassword
+  }
+}
+
+resource postgresWebapiAdminSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: 'ohdsi-admin-password'
+  parent: keyVault
+  properties: {
+    value: postgresWebapiAdminPassword
+  }
+}
+
+resource postgresWebapiAppSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: 'ohdsi-app-password'
+  parent: keyVault
+  properties: {
+    value: postgresWebapiAppPassword
+  }
+}
+
+resource jdbcConnectionStringWebapiAdminSecret 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = {
+  name: 'jdbc-connectionstring'
+  parent: keyVault
+  properties: {
+    value: jdbcConnectionStringWebapiAdmin
+  }
+}
+
+// Create a PostgreSql server
+resource postgresServer 'Microsoft.DBforPostgreSQL/flexibleServers@2022-12-01' = {
+  name: 'postgresql-${odhsiWebApiName}'
```

Comment:
What are the uniqueness requirements for this? Perhaps we should add a random element here?

Reply:
As we discussed with @guybartal - the naming convention now is '{name of the azure resource}-{id-given-by-the-user}-{generated suffix}'